examples/tutorials/api/nested_sampling_Crab.py

```diff
@@ -0,0 +1,415 @@
+# visulisation <https://johannesbuchner.github.io/UltraNest/method.html#visualisation>`__.
+# And for a tutorial of UltraNest applied to X-ray fitting with concrete examples and questions see : `BXA
+# Tutorial <https://peterboorman.com/tutorial_bxa.html>`__.
+#
+
+
+######################################################################
+# Proposed approach
+# -----------------
+#
+# In this example we will perform a Bayesian analysis with multiple 1D
+# spectra of the Crab nebula data and investigate their posterior
+# distributions.
+#
+
+
+######################################################################
+# Setup
+# -----
+#
+# As usual, we’ll start with some setup …
+#
+
+import matplotlib.pyplot as plt
+import numpy as np
+from gammapy.datasets import Datasets
+from gammapy.datasets import SpectrumDatasetOnOff
+
+from gammapy.modeling.models import (
+    SkyModel,
+    UniformPrior,
+    LogUniformPrior,
+)
+
+from gammapy.modeling.sampler import Sampler
+
+
+######################################################################
+# Loading the spectral datasets
+# -----------------------------
+#
+# Here we will load a few Crab 1D spectral data for which we will do a
+# fit.
+#
+
+path = "$GAMMAPY_DATA/joint-crab/spectra/hess/"
+
+datasets = Datasets()
+for id in ["23526", "23559", "23592"]:
+    dataset = SpectrumDatasetOnOff.read(f"{path}pha_obs{id}.fits")
+    datasets.append(dataset)
+
+
+######################################################################
+# Model definition
+# ----------------
+#
+# Now we want to define the spectral model that will be fitted to the
+# data.
+# The Crab spectra will be fitted here with a simple powerlaw for
+# simplicity.
+#
+
+model = SkyModel.create(spectral_model="pl")
+
+
+######################################################################
+#
+# .. WARNING:: Priors definition:
+#    Unlike a traditional fit where priors on the
+#    parameters is optional, here it is inherent to the Bayesian approach and
+#    are therefore mandatory.
+#
+# In this case we will set (min,max) prior that will define the
+# boundaries in which the sampling will be performed.
+# Note that it is usually recommended to use a `~gammapy.modeling.models.LogUniformPrior` for
+# the parameters that have a large amplitude range like the
+# `amplitude` parameter.
+# A `~gammapy.modeling.models.UniformPrior` means that the samples will be drawn with uniform
+# probability between the (min,max) values in the linear or log space
+# in the case of a `~gammapy.modeling.models.LogUniformPrior`.
+#
+
+model.spectral_model.amplitude.prior = LogUniformPrior(min=1e-12, max=1e-10)
+model.spectral_model.index.prior = UniformPrior(min=1, max=5)
+datasets.models = [model]
+datasets.models
+
+
+######################################################################
+# Defining the sampler and options
+# --------------------------------
+#
+# As for the `Fit` object, the `Sampler` object can receive
```

Comment:
The object `Sampler` is not visible in the API documentation, which is a problem when exposing this tutorial.
One should expose the `Sampler` object, and add  a proper documentation link.

Reply:
why is it not visible and how could this be solved ?

Reply:
If I do not mistaken, you should add in `__init.py__` of modeling the 3 classes of the `sampler.py` class. This should work...
And then, you can add in your python tuto `Sampler` -> `~gammapy.modeling.Sampler`

Reply:
it worked. Thx